pkg/gardenlet/operation/botanist/botanist.go

```diff
@@ -252,3 +256,103 @@ func (b *Botanist) RequiredExtensionsReady(ctx context.Context) error {
+				if oldPoolHash, ok := worker.Status.InPlaceUpdates.WorkerPoolToHashMap[oldPool.Name]; ok {
+					var (
+						kubernetesVersion    = b.Shoot.GetInfo().Spec.Kubernetes.Version
+						kubeletConfiguration = b.Shoot.GetInfo().Spec.Kubernetes.Kubelet
+					)
+
+					if pool.Kubernetes != nil {
+						if pool.Kubernetes.Version != nil {
+							kubernetesVersion = *pool.Kubernetes.Version
+						}
+
+						if pool.Kubernetes.Kubelet != nil {
+							kubeletConfiguration = pool.Kubernetes.Kubelet
+						}
+					}
+
+					newPoolHash, err := gardenerutils.CalculateWorkerPoolHashForInPlaceUpdate(
+						pool.Name,
+						&kubernetesVersion,
+						kubeletConfiguration,
+						ptr.Deref(pool.Machine.Image.Version, ""),
+						b.Shoot.GetInfo().Status.Credentials,
+					)
+					if err != nil {
+						return fmt.Errorf("failed to calculate worker pool %q hash: %w", pool.Name, err)
+					}
+
+					if oldPoolHash == newPoolHash {
+						continue
+					}
+				}
+			}
+		}
+
+		switch ptr.Deref(pool.UpdateStrategy, "") {
+		case gardencorev1beta1.AutoInPlaceUpdate:
+			autoInPlaceUpdatePendingWorkers.Insert(pool.Name)
+		case gardencorev1beta1.ManualInPlaceUpdate:
+			manualInPlaceUpdatePendingWorkers.Insert(pool.Name)
+		}
+	}
+
+	if autoInPlaceUpdatePendingWorkers.Len() == 0 && manualInPlaceUpdatePendingWorkers.Len() == 0 {
+		return nil
+	}
+
+	if b.Shoot.GetInfo().Status.InPlaceUpdates != nil &&
+		b.Shoot.GetInfo().Status.InPlaceUpdates.PendingWorkerUpdates != nil &&
+		autoInPlaceUpdatePendingWorkers.Equal(sets.New(b.Shoot.GetInfo().Status.InPlaceUpdates.PendingWorkerUpdates.AutoInPlaceUpdate...)) &&
+		manualInPlaceUpdatePendingWorkers.Equal(sets.New(b.Shoot.GetInfo().Status.InPlaceUpdates.PendingWorkerUpdates.ManualInPlaceUpdate...)) {
+		return nil
+	}
+
+	return b.Shoot.UpdateInfoStatus(ctx, b.GardenClient, true, func(shoot *gardencorev1beta1.Shoot) error {
+		if shoot.Status.InPlaceUpdates == nil {
+			shoot.Status.InPlaceUpdates = &gardencorev1beta1.InPlaceUpdatesStatus{}
+		}
+
+		if shoot.Status.InPlaceUpdates.PendingWorkerUpdates == nil {
+			shoot.Status.InPlaceUpdates.PendingWorkerUpdates = &gardencorev1beta1.PendingWorkerUpdates{}
+		}
+
+		for poolName := range autoInPlaceUpdatePendingWorkers {
+			if slices.Contains(shoot.Status.InPlaceUpdates.PendingWorkerUpdates.AutoInPlaceUpdate, poolName) {
+				continue
+			}
+			shoot.Status.InPlaceUpdates.PendingWorkerUpdates.AutoInPlaceUpdate = append(shoot.Status.InPlaceUpdates.PendingWorkerUpdates.AutoInPlaceUpdate, poolName)
+		}
+
+		for poolName := range manualInPlaceUpdatePendingWorkers {
+			if slices.Contains(shoot.Status.InPlaceUpdates.PendingWorkerUpdates.ManualInPlaceUpdate, poolName) {
+				continue
+			}
+			shoot.Status.InPlaceUpdates.PendingWorkerUpdates.ManualInPlaceUpdate = append(shoot.Status.InPlaceUpdates.PendingWorkerUpdates.ManualInPlaceUpdate, poolName)
+		}
```

Comment:
Since iteration over map is not deterministic, will this result in a different ordering again and again?

Reply:
Nice catch, I changed it to a slice instead of set, since the worker names shouldn't repeat themselves anyway, and added a test case for this.